services/src/modules/directives/rest/datasource.ts

```diff
@@ -86,10 +89,16 @@ export class RESTDirectiveDataSource extends RESTDataSource<RequestContext> {
     args: GraphQLArguments,
     info: GraphQLResolveInfo
   ) {
-    if (!kvs || kvs.length == 0) return;
+    let validQueryParams = true;
+
+    if (!kvs || kvs.length == 0) return validQueryParams;
 
     for (const kv of kvs) {
       const value = inject(kv.value, parent, args, this.context, info);
+      if (!value && kv.required) {
+        validQueryParams = false;
```

Comment:
If required query param is missing the function should throw `GraphQLError`.
The field in response will be `null` and there will be error in `errors` filed.
```suggestion
        throw new GraphQLError(`${kv.key} query parameter is required`);
```

Reply:
There is a valid scenario which causes us to not want to add this throw error here. Which is the use case I am trying to solve for with this PR. Using the rest directive on a field instead of a query, means that there are some cases when the source might not have a value that is being used as input to the rest directive. For example, I am using @rest to resolve a billing field on an agreement (parent type). Some agreements do not have billing ids associated with it (ie, client is handling billing vs us). In those cases the source record will not return billing ids which is what @rest uses as input for resolving this field. That doesn't mean its wrong, it just doesn't have enough source data to resolve the field. What are your thoughts around this scenario?

Reply:
Additional note in general on this - I added the throw statement, however, the initial use case I was trying to account for still ends up sending the request since I am using argument injection to resolve the input query parameter. If there is no match, its returning whatever the input was. The api request sends with argument injection as the value for the query parameter. Any thoughts on how to get around this?

Reply:
Following our discussion in Slack.
1) The first point will be resolved by adding optional parameter `condition` to `@localResolver` directive.
2) The second point can be resolved by changing the `value` expression to return `null` in the right cases. For example, instead of `"{source.foo.bar}"` when `foo` may be null you can set `"{source.foo && source.foo.bar}"`.
In this way you'll avoid an exception in argument evaluation and the return value will be `null`